cli/src/build-settings.ts

```diff
@@ -57,7 +57,11 @@ export default class BuildSettings {
       npm_config_disturl:           process.env.npm_config_disturl || null,
       npm_config_runtime:           process.env.npm_config_runtime || null,
       npm_config_build_from_source: process.env.npm_config_build_from_source || null,
-      npm_config_devdir:            process.env.npm_config_devdir || null
+      npm_config_devdir:            process.env.npm_config_devdir || null,
+      npm_config_node_engine:       process.env.npm_config_node_engine || null,
+      npm_config_nodedir:           process.env.npm_config_nodedir || null,
+      npm_config_node_gyp:          process.env.npm_config_node_gyp || null,
+      npm_config_platform:          process.env.npm_config_platform || null
```

Comment:
Adding these values is going to force rebuilds for users when they upgrade the CLI, but considering we recommend keeping the CLI and `neon` versions in sync, most users would need to rebuild anyway. 👍